src/main/java/spark/route/RouteEntry.java

```diff
@@ -95,33 +97,28 @@ private boolean matchPath(String path) { // NOSONAR
         } else {
             // Number of "path parts" not the same
             // check wild card:
-            if (this.path.endsWith("*")) {
-                if (pathSize == (thisPathSize - 1) && (path.endsWith("/"))) {
-                    // Hack for making wildcards work with trailing slash
-                    pathList.add("");
-                    pathList.add("");
-                    pathSize += 2;
-                }
+            if (this.path.contains("*") && thisPathSize < pathSize) {
+                StringBuilder sb = new StringBuilder();
+                char[] charArray = this.path.toCharArray();
+                for (int i = 0 ; i < charArray.length ; ++i) {
+                    if (charArray[i] == '*') {
+                        sb.append('.');
+                    }
 
-                if (thisPathSize < pathSize) {
-                    for (int i = 0; i < thisPathSize; i++) {
-                        String thisPathPart = thisPathList.get(i);
-                        String pathPart = pathList.get(i);
-                        if (thisPathPart.equals("*") && (i == thisPathSize - 1) && this.path.endsWith("*")) {
-                            // wildcard match
-                            return true;
-                        }
-                        if (!thisPathPart.startsWith(":")
-                                && !thisPathPart.equals(pathPart)
-                                && !thisPathPart.equals("*")) {
-                            return false;
+                    if (charArray[i] == ':') {
+                        while (charArray[++i] != '/') {
```

Comment:
this would match multiple path parts if you set :some_variable. I don't think that's the expected behavior.
for example /some/:test/match would match /some/stuff/that/shouldnt/match

Reply:
you can probably use something like [^/]* instead of .*

Reply:
fine, then /some/:test/match/* and /some/stuff/that/shouldnt/match/magic :)

Reply:
thank you for breaking it :D

Reply:
btw, I implemented the backtracking solution for fun :)
```java
private boolean matchPath(List<String> candidates, List<String> path, int i, int j) {
        if (i == candidates.size() && j == path.size()) return true;
        if (i == candidates.size() || j == path.size()) return false;
        
        String candidatePart = candidates.get(i);
        String pathPart = path.get(j);

        if (pathPart.startsWith(":")) {
            return matchPath(candidates, path, i + 1, j + 1);
        } else if (pathPart.equals("*")) {
            for (int k = i + 1; k < candidatePart.length(); k++) {
                if (matchPath(candidates, path, k, j + 1)) {
                    return true;
                }
            }
            return false;
        } else if (candidatePart.equals(pathPart)) {
            return matchPath(candidates, path, i + 1, j + 1);
        } else {
            return false;
        }
    }
```

Reply:
thank you!